src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/UnusedTagsPass.php

```diff
@@ -59,7 +59,7 @@ public function process(ContainerBuilder $container)
 
         foreach ($container->findUnusedTags() as $tag) {
             // skip whitelisted tags
-            if (in_array($tag, $this->whitelist)) {
+            if (empty($tag) || in_array($tag, $this->whitelist)) {
```

Comment:
this misses `'0'` as well. IMO an empty tag should still be reported as "unused". only the `$candidates` calculation should be skipped for empty tags.